### PR TITLE
fix: update_form_change operation field would not update from `archive` to `update`

### PR DESCRIPTION
**Pull request by gurjmatharu**
Fix for #1068

### CODE REVIEW

**Review by tmastrom — COMMENTED** [review]
We can now remove the workaround in [1066](https://github.com/bcgov/cas-cif/pull/1062/files)

Also it could be worthwhile to write a test confirming form_change record with a 'create' operation will NOT automatically be changed to 'update'

File: schema/test/unit/util_functions/import_swrs_operators_test.sql [hunk 1]
```diff
@@ -51,7 +51,7 @@ select results_eq(
       (new_form_data->'swrsOrganisationId')::int,
       operation::text,
       change_status::text
-    from cif.form_change where form_data_table_name='operator'
+    from cif.form_change where form_data_table_name='operator' order by id
```
Comment (BCerki) [inline schema/test/unit/util_functions/import_swrs_operators_test.sql:54]: I wonder why the changes affected this test

File: schema/deploy/mutations/update_form_change.sql [hunk 2]
```diff
@@ -9,8 +9,10 @@ $$
 
   update cif.form_change set
     new_form_data = coalesce(form_change_patch.new_form_data, new_form_data),
-    operation = coalesce(form_change_patch.operation, operation),
-    validation_errors = coalesce(form_change_patch.validation_errors, validation_errors),
+    operation = case
+      when form_change_patch.operation is null and operation = 'archive' then 'update'
+      else coalesce(form_change_patch.operation, operation)
+    end,    validation_errors = coalesce(form_change_patch.validation_errors, validation_errors),
```
Comment (BCerki) [inline schema/deploy/mutations/update_form_change.sql:15]: I don't totally understand what the card is asking for, so I'm not sure if this change covers everything. In this sentence:
> When the update_form_change() mutation is passed an operation as part of the patch, then it will update the operation to 'update'. When not passed 'update' as the operation, the coalesce in the mutation defaults back to 'archive'.

Is that saying that regardless of what operation `update_form_change` receives, it will always default to "update"? In which case I think that needs addressing?

File: schema/test/unit/util_functions/import_swrs_operators_test.sql (hunk 1 above)
Comment (gurjmatharu) [reply schema/test/unit/util_functions/import_swrs_operators_test.sql:54]: > If sorting is not chosen, the rows will be returned in an unspecified order. The actual order in that case will depend on the scan and join plan types and the order on disk, but it must not be relied on. A particular output ordering can only be guaranteed if the sort step is explicitly chosen.

Not sure what exactly happened but it's something we can't rely on, best practice should be to include an `order by` clause.

File: schema/deploy/mutations/update_form_change.sql (hunk 2 above)
Comment (gurjmatharu) [reply schema/deploy/mutations/update_form_change.sql:15]: My understanding is that quote is describing current behaviour.  Passing in `UPDATE` or `undefined`/`null` provide different results.

The expected behaviour is that regardless of explicitly stating we want an `UPDATE` or passing in nothing we should get the same result, which is an `UPDATE`.

This is due to the `coalesce` where the first parameter will be `NULL` and second will be `ARCHIVE` so we need to conditionally check for this unique case.

File: app/components/Form/ProjectManagerFormGroup.tsx [hunk 3]
```diff
@@ -137,8 +136,7 @@ const ProjectManagerFormGroup: React.FC<Props> = (props) => {
           rowId,
           formChangePatch: {
             newFormData,
-            // a temporary fix to make sure the form change operation is set correctly
-            operation: projectRevision.isFirstRevision ? "CREATE" : "UPDATE",
+            operation: undefined,
```
Comment (dleard) [inline app/components/Form/ProjectManagerFormGroup.tsx:139]: we can remove this operation line now

File: app/components/Form/ProjectManagerFormGroup.tsx [hunk 4]
```diff
@@ -148,7 +146,7 @@ const ProjectManagerFormGroup: React.FC<Props> = (props) => {
             id: formChangeId,
             changeStatus: "pending",
             newFormData,
-            operation: projectRevision.isFirstRevision ? "CREATE" : "UPDATE",
+            operation: undefined,
```
Comment (dleard) [inline app/components/Form/ProjectManagerFormGroup.tsx:149]: same here, this operation line can be removed

**Review by dleard — CHANGES_REQUESTED** [review]
Looks great! 
I tested it and the mutation is properly assigning update to archived records when appropriate.
Just a couple unnecessary lines to clean up and then should be good to go

**Review by dleard — APPROVED** [review]
